modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java

```diff
@@ -2949,43 +2976,48 @@ public VisorGatewayArgument forNodes(Collection<ClusterNode> nodes) {
         }
 
         /**
-         * Add custom argument.
+         * Add custom arguments.
          *
-         * @param vals Values.
+         * @param vals Array of values or {@code null}.
          * @return This helper for chaining method calls.
          */
-        public VisorGatewayArgument arguments(Object... vals) {
-            for (Object val : vals)
-                put("p" + idx++, String.valueOf(val));
+        public VisorGatewayArgument addArguments(@Nullable Object... vals) {
+            if (idx == 3)
+                throw new IllegalStateException("Task argument class should be declared before adding of additional arguments. " +
+                    "Use VisorGatewayArgument.setTaskArgument first");
 
-            return this;
-        }
+            if (vals != null && F.isEmpty(vals))
```

Comment:
Change to `if (F.isEmpty(vals)) ....` and move inside next if block.

Reply:
null value of vararg argument is not array with null value. It is null reference to array object.